src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import akka.actor.SupervisorStrategy.Restart
+import akka.actor.{Actor, ActorRef, OneForOneStrategy, Props}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.config.MonitorConfig
+import org.broadinstitute.dsde.workbench.leonardo.dao.DataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterRequest}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.ClusterMonitorSupervisor._
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object ClusterMonitorSupervisor {
+  def props(monitorConfig: MonitorConfig, gdDAO: DataprocDAO, dbRef: DbReference): Props =
+    Props(new ClusterMonitorSupervisor(monitorConfig, gdDAO, dbRef))
+
+  sealed trait ClusterSupervisorMessage
+  case class ClusterCreated(cluster: Cluster) extends ClusterSupervisorMessage
+  case class ClusterDeleted(cluster: Cluster, recreate: Boolean = false) extends ClusterSupervisorMessage
+  case class RecreateCluster(cluster: Cluster) extends ClusterSupervisorMessage
+}
+
+class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, gdDAO: DataprocDAO, dbRef: DbReference) extends Actor with LazyLogging {
+  import context.dispatcher
+
+  override def receive: Receive = {
+    case ClusterCreated(cluster) =>
+      logger.info(s"Monitoring cluster ${cluster.googleProject}/${cluster.clusterName} for initialization.")
+      startClusterMonitorActor(cluster)
+
+    case ClusterDeleted(cluster, recreate) =>
+      logger.info(s"Monitoring cluster ${cluster.googleProject}/${cluster.clusterName} for deletion.")
+      startClusterMonitorActor(cluster, recreate)
+
+    case RecreateCluster(cluster) =>
```

Comment:
can we move the recreation logic elsewhere? doesn't feel right for the supervisor to be doing this work. doesn't this duplicate the guts of `LeoService.createCluster`?

Reply:
+1

Reply:
Sure, will think about it. I didn't want to have the supervisor calling `LeoService` because that would create a circular dependency. But maybe there is some common place where this logic could live.

Reply:
Solved by adding a `createCluster` method in the `LeoService` object.

Reply:
Was able to call `leoService` directly by registering it with this actor, see below.

I tried another way of having a circular dependency using lazy val's, but it felt ugly.